shared/reducers/profile.js

```diff
@@ -9,7 +9,7 @@ export default function (state = initialState, action) {
 
   switch (action.type) {
     case CommonConstants.resetStore:
-      return {...initialState}
+      return Immutable.fromJS({...initialState.toJS()})
```

Comment:
so react native doens't do object assign perfectly so it balks at this kind of stuff